### PR TITLE
DOC: fix clustering example

### LINKED ISSUE #520: MeanInterbuildingDistance outputs Nans
### Describe the problem

The Momepy function MeanInterbuildingDistance outputs a columns full of Nans even though no error is shown.  

### Steps to reproduce

Replicating the example provided in http://docs.momepy.org/en/stable/examples/clustering.html 

This is the line that outputs the unexpected result:

buildings['interbuilding_distance'] = momepy.MeanInterbuildingDistance(buildings, queen_1, 'uID', queen_3, verbose=False).series

### Versions of your packages

momepy=0.6.0
geopandas=0.14.1 
pysal=23.1
python=3.10.13

### Your operating system

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by martinfleis**
Closes #520

The diff is terrible but this is the change:

```diff

- buildings['interbuilding_distance'] = momepy.MeanInterbuildingDistance(buildings, queen_1, 'uID', queen_3, verbose=False).series
+ buildings['interbuilding_distance'] = momepy.MeanInterbuildingDistance(buildings, queen_1, 'uID', 3, verbose=False).series
```

The code in the example was just wrong, resulting in Exceptions when creating the ego graph, resulting in NaNs.

### CODE REVIEW

**Comment by martinfleis** [comment]
I also don't like the try/except implementation we have in there but have no energy to refactor that. It is not buggy but just a bad pattern. I also think that with `libpysal.graph`, I would be able to vectorise the whole thing but that is a 2024 problem.

**Comment by jGaboardi** [comment]
> I also don't like the try/except implementation we have in there but have no energy to refactor that. It is not buggy but just a bad pattern. I also think that with `libpysal.graph`, I would be able to vectorise the whole thing but that is a 2024 problem.

Yeah, let's make an independent issue for that and punt.

File: docs/examples/clustering.ipynb
Comment (jGaboardi) [inline docs/examples/clustering.ipynb:1]: Thrown warnings – catch, fix, or do nothing:

* cell 1
    ```
    /Users/martin/miniforge3/envs/pysal/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress
    not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
        from .autonotebook import tqdm as notebook_tqdm
    ```
   * Should we add a warning catcher for this? Or fix it?
* cell 4
    ```
    /var/folders/2f/fhks6w_d0k556plcv3rfmshw0000gn/T/ipykernel_81241/1465348557.py:1: UserWarning: The
    `geometries` module and `geometries_from_X` functions have been renamed the `features` module and
    `features_from_X` functions. Use these instead. The `geometries` module and function names are deprecated             
    and will be removed in a future release.
        buildings = osmnx.geometries.geometries_from_place(place, tags={'building':True})
    ```
    * Same here
* cell 11
    ```
    /Users/martin/miniforge3/envs/pysal/lib/python3.11/site-packages/geopandas/array.py:1459: UserWarning: CRS
    not set for some of the concatenation inputs. Setting output's CRS as S-JTSK / Krovak East North (the single
    non-null crs provided).
        return GeometryArray(data, crs=_get_common_crs(to_concat))
    ```     
    * Same here
* cell 13
    ```
    /var/folders/2f/fhks6w_d0k556plcv3rfmshw0000gn/T/ipykernel_81241/1328706492.py:3: UserWarning: 
    Tessellation does not fully match buildings. 20 element(s) collapsed during generation - unique_id: {3968, 
    4227, 11255, 10686, 10687, 9026, 4177, 11473, 11474, 8534, 11478, 4185, 4188, 10465, 9061, 3958, 4215, 
    4218, 4222, 4223}.
        tessellation = momepy.Tessellation(buildings, "uID", limit, verbose=False, segment=1)
    /var/folders/2f/fhks6w_d0k556plcv3rfmshw0000gn/T/ipykernel_81241/1328706492.py:3: UserWarning: 
    Tessellation contains MultiPolygon elements. Initial objects should  be edited. `unique_id` of affected 
    elements: [4167, 3227, 11783, 3173, 10997].
        tessellation = momepy.Tessellation(buildings, "uID", limit, verbose=False, segment=1)
   * We'll probably want to keep these in to demonstrate the "normalcy" of receiving that warning.
* cell 20
    ```
    /Users/martin/Git/momepy/momepy/distribution.py:137: FutureWarning: The `query_bulk()` method 
    is deprecated and will be removed in GeoPandas 1.0. You can use the `query()` method instead.
        inp, res = gdf.sindex.query_bulk(gdf.geometry, predicate="intersects")
    /Users/martin/miniforge3/envs/pysal/lib/python3.11/site-packages/shapely/set_operations.py:131: 
    RuntimeWarning: invalid value encountered in intersection
        return lib.intersection(a, b, **kwargs)
    ```
    * Should we add a warning catcher for this? Or fix it?
* cell 24
    ```
    /var/folders/2f/fhks6w_d0k556plcv3rfmshw0000gn/T/ipykernel_81241/738459904.py:2: FutureWarning: 
    `use_index` defaults to False but will default to True in future. Set True/False directly to control this behavior 
    and silence this warning
        buildings_q1 = libpysal.weights.contiguity.Queen.from_dataframe(buildings, silence_warnings=True)
    ```
    * Same here

Comment (martinfleis) [reply docs/examples/clustering.ipynb:1]: I've fixed the first two. The rest needs some refactoring.

Comment (jGaboardi) [reply docs/examples/clustering.ipynb:1]: > I've fixed the first two. The rest needs some refactoring.

Those I'd say let's punt till next time.

Comment (jGaboardi) [inline docs/examples/clustering.ipynb:1]: Now we have 2 new ones in cells 29 & 31:

```
/Users/martin/miniforge3/envs/stable/lib/python3.12/site-packages/tqdm/std.py:580: DeprecationWarning:
datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use 
timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, 
datetime.UTC).
    if rate and total else datetime.utcfromtimestamp(0))
```

Either ignore or catch. Refactor can wait for future.

Comment (jGaboardi) [reply docs/examples/clustering.ipynb:1]: If we want to ignore, I'd say this PR is ready go so I'll go ahead and hit approve.

**Review by jGaboardi — APPROVED** [review]

File: docs/examples/clustering.ipynb
Comment (martinfleis) [reply docs/examples/clustering.ipynb:1]: That's on tqdm to fix.

**Pull request merged**